src/status_providers/github.ts

```diff
@@ -254,6 +283,34 @@ export class GithubStatusProvider extends BaseStatusProvider {
     return revisionStatus;
   }
 
+  /**
+   * Gets revision checks from GitHub Check runs API
+   *
+   * API docs:
+   * https://developer.github.com/v3/checks/suites/#list-check-suites-for-a-git-reference
```

Comment:
Note that this requires the `checks:read` permission. I believe, we already require the generic `read` permission, so this shouldn't pose as an issue.